Modified Suleiman-Ritchie soil evaporation

chporter opened the pull request:
Modified as per Ayman Suleiman at Jan 2025 DSSAT sprint.
-- ES occurs from top 15 cm only. Re-instate UPFLOW. 
-- Uses MESEV = "M" (for Modified Suleiman-Ritchie)

This runs for all DSSAT experiments and treatments. Soil evaporation is considerably lower than all other methods. I ran the default soil evaporation in each FileX vs R, S, and M method override using external simulation control file.

Average ratio over all experiments / treatments
0.98393 | Ritchie : default
1.183594 | Suleiman-Ritchie : default
0.716105 | Modified Suleiman-Ritchie : default
0.62579 | Modified Suleiman-Ritchie : Suleiman-Ritchie

This needs to be thoroughly vetted by Ayman, Gerrit, Fabio, Bruce before accepting the pull request.

Commit:
Proposed change to Suleiman-Ritchie soil evaporation
- also needs to re-implement UPFLOW

(cherry picked from commit 27ab1296d18ca43765c1963c460219ec9247ce7f)

Commit:
ESR modifications unfinished

Commit:
ESR_SoilEvap_mod - complete, needs testing

Commit:
bug fixes - runs for all DSSAT experiments / treatments

Commit:
clean up notes in old S-R routine

Commit:
Work on VBOSE problem.
- this solves much of the problems with Bambara Groundnut, but there are still some small differences
- CO2EM in Summary.OUT problem fixed.

Commit:
In IPSIM.for, move all switch and method code checking to after external simulation controls are read
- previously, we had redundant code with separate checks for FileX simulation controls and for external SC
- caused a problem with guar, which was allowed for N fixation in FileX but not external SC.

Commit:
Remove some debugging code

Commit:
ALOHA bug fix for VBOSE = 0

Commit:
Remove checks from External_SimControls

Commit:
SAVE MEPHO, ISWSYM switches when using external controls

Commit:
IPSIM - fix for ISWSYM

Commit:
Write the daily output for GHG.OUT file

aymansuleiman commented:
Dear Mrs. Porter,
Many thanks for your trying the soil evaporation from the 15 cm depth only.  I suggest that we try soil evaporation from 30 cm instead from om 15 because the simulated SOIL EVAPORATION WAS low.
Kind regards,
Ayman


    On Tuesday, February 25, 2025 at 07:12:40 PM EST, Cheryl Porter ***@***.***> wrote:  
 
 


Closed #471.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because your review was requested.Message ID: ***@***.***>

Commit:
Merge branch 'develop' of https://github.com/dssat/dssat-csm-os into Sul-Rit2

chporter commented:
@aymansuleiman @GerritHoogenboom 
I changed the soil depth to 30 cm and it improved results (i.e., came closer to what we expected), but still only evaporates about 86% on average of Ritchie method and about 72% of the original Suleiman-Ritchie method. 
[ET_comparison_30.xlsx](https://github.com/user-attachments/files/19045518/ET_comparison_30.xlsx)

Commit:
Merge branch 'vbose' of https://github.com/chporter/dssat-csm-os into Sul-Rit2

Commit:
Suleiman-Ritchie method uses 30 cm to limit soil evaporation (instead of 15cm)

aymansuleiman commented:
Many thanks for changing the soil depth to 30 cm. For sure we need to change the depth one more time. Please, can you change the depth to 45 m.


    On Sunday, March 2, 2025 at 07:31:12 PM EST, Cheryl Porter ***@***.***> wrote:  
 
  
@aymansuleiman @GerritHoogenboom
I changed the soil depth to 30 cm and it improved results (i.e., came closer to what we expected), but still only evaporates about 86% on average of Ritchie method and about 72% of the original Suleiman-Ritchie method.
ET_comparison_30.xlsx

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
chporter left a comment (DSSAT/dssat-csm-os#471)
@aymansuleiman @GerritHoogenboom
I changed the soil depth to 30 cm and it improved results (i.e., came closer to what we expected), but still only evaporates about 86% on average of Ritchie method and about 72% of the original Suleiman-Ritchie method.
ET_comparison_30.xlsx

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Commit:
Modify Modified Suleiman-Ritchie soil evap method
- don't export SWDELTU because we also call UPFLOW with this method.
- add some intializations
- make more generic, can easily change depths of soil evaporation by changing one line of code.
- currently using the same A and B coefficients as the original for comparison. Will change these in a separate step after testing.

Commit:
remove ratio of diffusion methods
- not needed or used anywhere
- causes zero divide

Commit:
Mod SR - use Ayman Suleiman's A & B coefficients from the paper
- final depth of evaporation = 50 cm
- this can be changed easily by modifying line 60

chporter commented:
I modified the method with the following:
- it's now more generic and the depth of evap can be changed with one line of code to simplify exploring the range of options.
- Modified S-R calls UPFLOW routine to move water up, but also takes water from layers up to specified depth. The evaporation is not (yet) added to UPFLOW which is used to move soil N. We probably want to make this change before finalizing it.
- I corrected an error where the mulch evaporation was not taken into account prior to calling the routine previously, so these results are slightly different.

The 15cm depth and even going up to 30 cm is not enough to increase evaporation to more than the Ritchie method (on average). I also tried 50 cm and it does increase evaporation slightly (+4% over the Ritchie method). 

I also tested the model with a depth of 200 cm which should give similar results to the original Suleiman-Ritchie model. It is very similar, with differences due to the addition of UPFLOW.

The attached plot shows how the various methods compare to the Ritchie method for all the DSSAT experiments and treatments.

The source code currently has the 50 cm depth but we need to discuss this prior to merging the pull request.

![image](https://github.com/user-attachments/assets/4b8d9671-d4a1-408f-b829-219ecdc30690)

aymansuleiman commented:
Many thanks for you traying the 50 cm depth. I agree with you that 15 and 30 cm depths are not suitable because their evaporation is lower than Ritch method. 50 cm depth is closer as it evaporates on average about 90% of Ritchie method. Using the 50 cm is more appropriate.  
    On Friday, March 14, 2025 at 12:18:18 AM EDT, Cheryl Porter ***@***.***> wrote:  
 
  
I modified the method with the following:
   
   - it's now more generic and the depth of evap can be changed with one line of code to simplify exploring the range of options.
   - Modified S-R calls UPFLOW routine to move water up, but also takes water from layers up to specified depth. The evaporation is not (yet) added to UPFLOW which is used to move soil N. We probably want to make this change before finalizing it.
   - I corrected an error where the mulch evaporation was not taken into account prior to calling the routine previously, so these results are slightly different.

The 15cm depth and even going up to 30 cm is not enough to increase evaporation to more than the Ritchie method (on average). I also tried 50 cm and it does increase evaporation slightly (+4% over the Ritchie method).

I also tested the model with a depth of 200 cm which should give similar results to the original Suleiman-Ritchie model. It is very similar, with differences due to the addition of UPFLOW.

The attached plot shows how the various methods compare to the Ritchie method for all the DSSAT experiments and treatments.

The source code currently has the 50 cm depth but we need to discuss this prior to merging the pull request.

image.png (view on web)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
chporter left a comment (DSSAT/dssat-csm-os#471)
I modified the method with the following:
   
   - it's now more generic and the depth of evap can be changed with one line of code to simplify exploring the range of options.
   - Modified S-R calls UPFLOW routine to move water up, but also takes water from layers up to specified depth. The evaporation is not (yet) added to UPFLOW which is used to move soil N. We probably want to make this change before finalizing it.
   - I corrected an error where the mulch evaporation was not taken into account prior to calling the routine previously, so these results are slightly different.

The 15cm depth and even going up to 30 cm is not enough to increase evaporation to more than the Ritchie method (on average). I also tried 50 cm and it does increase evaporation slightly (+4% over the Ritchie method).

I also tested the model with a depth of 200 cm which should give similar results to the original Suleiman-Ritchie model. It is very similar, with differences due to the addition of UPFLOW.

The attached plot shows how the various methods compare to the Ritchie method for all the DSSAT experiments and treatments.

The source code currently has the 50 cm depth but we need to discuss this prior to merging the pull request.

image.png (view on web)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
